Tutorial section docs (#13297)
PrefectHQ/prefect: docs/guides/automations.md

    list_of_names = build_names()
```

The data cleaning workflow has visibility into each step, and sends a list of names to the next step of the pipeline.

## Create notification block within the UI
Next, send a notification based off a completed state outcome. Configure a notification that shows when to look into our workflow logic. 

1. Prior to creating the automation, confirm the notification location. Create a notification block to help define where the notification is sent. 
![List of available blocks](/img/guides/block-list.png)

2. Navigate to the blocks page on the UI, and click into creating an email notification block. 
![Creating a notification block in the Cloud UI](/img/guides/notification-block.png)

3. Go to the automations page to create your first automation.
![Automations page](/img/guides/automation-list.png)

4. Next, find the trigger type. In this case, use a flow completion. 
![Trigger type](/img/guides/automation-triggers.png)

> 5. Finally, create the actions for when the trigger is hit. In this case, create a notification to showcase the completion. 

Reviewer: ```suggestion
5. Finally, specify one or more actions to be taken when the trigger condition is satisfied. In this case, create a notification to showcase the completion.
```

![Notification block in automation](/img/guides/notify-auto-block.png)

6. Now the automation is ready to be triggered from a flow run completion. Run the file locally and see that the notification is sent to your inbox after the completion. It may take a few minutes for the notification to arrive.
![Final notification](/img/guides/final-automation.png)

> !!! Tip "No deployment created"

Reviewer: We should remove this tip entirely

    return response.json()
```
 

> After running this function, you will see the changes that came from the post request within the UI. Keep in mind the context will be "custom" on UI. 

Reviewer: ```suggestion
After running this function, you will see the new automation with a "custom" in the UI.
```

> Run the underlying flow and see the deployment kick off after 30 seconds. This results in a new flow run of `build_names`. You can see this new deployment get initiated with the custom parameters outlined above. 

Reviewer: ```suggestion
Run the underlying flow and see the deployment kick off after 30 seconds. This results in a new flow run of `build_names` with the custom parameters provided above. 
```

> In a few quick changes, you can programmatically create an automation that deploys workflows with custom parameters. 

Reviewer: ```suggestion
With a few simple changes, you can programmatically create an automation that triggers flow runs with custom parameters. 
```

> ## Use an underlying .yaml file

Reviewer: ```suggestion
## Use an .yaml file
```

> You can take this a step further by using your own .yaml version of the automation, and registering that file with the UI. This simplifies the requirements of the automation by declaring it in its own .yaml file, and then registering that .yaml with the API. 

Reviewer: ```suggestion
You can take this a step further by specifying the automation in a .yaml file, and then registering that specification with the API.
```

> First start with creating the .yaml file to house the automation requirements:

Reviewer: ```suggestion
First start with creating the .yaml file to house the automation specification:
```

> ```yaml

Reviewer: ```suggestion
```yaml title="automation.yaml"
```

  - type: "cancel-flow-run"
```

> Make a helper function that applies this YAML file with the REST API function. 

Reviewer: ```suggestion
Make a helper function that applies this YAML file with the REST API. 
```

![webhook-simple](/img/guides/webhook-simple.png)

Each webhook corresponds to a custom event created, where you can react to it downstream with a separate deployment or automation. 

For example, you can create a curl request that sends the endpoint information such as a run count for your deployment. 
```console
curl -X POST https://api.prefect.cloud/hooks/34iV2SFke3mVa6y5Y-YUoA -d "model_id=adhoc" -d "run_count=10" -d "friendly_name=test-user-input"
```
From here, you can make a webhook that is connected to pulling in parameters on the curl command, and then it kicks off a deployment that uses these pulled parameters.
![Webhook created](/img/guides/webhook-created.png)

Go into the event feed to automate straight from this event. 
![Webhook automate](/img/guides/webhook-automate.png)

This allows you to create automations that respond to these webhook events. From a few clicks in the UI, you can associate an external process with the Prefect events API that can trigger downstream deployments. 
![Automation custom](/img/guides/automation-custom.png)

In the next section, you will explore event triggers that automate the kickoff of a deployment run.

## Use triggers

> You can take this a step further by creating a deployment that triggers when a flow run takes longer than expected. 

Reviewer: ```suggestion
You can go a step further by creating an automation that triggers a deployment when a flow run takes longer than expected. 
```

You can take advantage of Prefect's [Marvin](https://www.askmarvin.ai/) library that uses an LLM to classify the data. 
Marvin is great at embedding data science and data analysis applications within your pre-existing data engineering workflows. In this case, you can use [Marvin's AI functions](https://www.askmarvin.ai/docs/text/functions/) to help make the dataset more information rich. 

Install Marvin with `pip install marvin` and [set your OpenAI API key](https://www.askmarvin.ai/welcome/tutorial/).

You can add a trigger to run a deployment in response to a specific event. 

> Here's an example with Marvin's AI functions. It will take in a Pandas DataFrame and use the AI function to analyze it. 

Reviewer: ```suggestion
This example takes in a Pandas DataFrame and uses an AI function to analyze it. 
```

## Next steps

You've seen how to create automations with the UI, REST API, and a trigger defined in a `prefect.yaml` deployment definition.

> To learn more about events that can act as automation triggers, see the [events docs](/concepts/events/).  To learn more about event webhooks in particular, see the [webhooks guide](/guides/webhooks/).

Reviewer: ```suggestion
To learn more about Prefect events, which can trigger automations, see the [events docs](/concepts/events/). Prefect Cloud can receive external events webhooks. See the [webhooks guide](/guides/webhooks/) to learn how to create them.
```
